### PR TITLE
[r] Expand developer instructions in README.md

### PR DESCRIPTION
**Pull request by eddelbuettel**
**Issue and/or context:**

The R README does not currently detail installation from source in the same detail as Python and C++

**Changes:**

Expanded README.md, tested locally.

**Notes for Reviewer:**

#1032
[SC 25993](https://app.shortcut.com/tiledb-inc/story/25993/add-developer-information-to-the-wiki)

### CODE REVIEW

File: apis/r/README.md [hunk 1]
```diff
@@ -14,13 +14,26 @@ This code is hosted at R-universe, so you can do
...
+* This, and the other R package dependencies, are listed in the [DESCRIPTION](https://github.com/single-cell-data/TileDB-SOMA/blob/main/apis/r/DESCRIPTION) file and can be installed via _e.g_ `remotes::install_dep(".")`.
+* In addition, the R package also depends on the [`libtiledbsoma` library](https://github.com/single-cell-data/TileDB-SOMA/tree/main/libtiledbsoma) -- which gets installed with the package as
+  described in the next section.
+
+### Step-by-step
+
+* Clone this repository: `git clone git@github.com:single-cell-data/TileDB-SOMA.git`
+* Change into the R API package directory: `cd TileDB-SOMA/apis/r`
+* Optionally, clean the file in to the repo: `./cleanup` (this is not needed the first time)
+* Optionally, update the `libtiledbsoma` sources: `./copy_source.sh` (which updates the includes tarball of `libtiledbsoma`).
```
Comment (johnkerl) [inline apis/r/README.md:31]: perhaps it would make sense to put the `remotes::install_deps(".")` bit here -- ?

**Review by johnkerl — COMMENTED** [review]
Still doing dry run on a fresh EC2 instance, but, a couple quick comments above ...

File: apis/r/README.md [hunk 2]
```diff
@@ -14,13 +14,26 @@ This code is hosted at R-universe, so you can do
...
 
-Note that building from source (using `R CMD INSTALL .`) implies also building
-[libtiledbsoma](https://github.com/single-cell-data/TileDB-SOMA/tree/main/libtiledbsoma)
-from source so additional requirements listed in its
-[README.md](https://github.com/single-cell-data/TileDB-SOMA/blob/main/libtiledbsoma/README.md)
-may apply.
+### Requirements
+
+* Source installation requires the [`tiledb` R package](https://github.com/TileDB-Inc/TileDB-R) (which in turn depends on the [`tiledb` Core library](https://github.com/TileDB-Inc/TileDB)).
+* This, and the other R package dependencies, are listed in the [DESCRIPTION](https://github.com/single-cell-data/TileDB-SOMA/blob/main/apis/r/DESCRIPTION) file and can be installed via _e.g_ `remotes::install_dep(".")`.
```
[the remark this answers is not in the document]
Comment (eddelbuettel) [reply apis/r/README.md:22]: Good catch. As I use a wrapper, or the command in CI, I never type it.

File: apis/r/README.md (hunk 1 above)
Comment (eddelbuettel) [reply apis/r/README.md:31]: Dunno we have in the preceding para. I omitted it on purpose.

File: apis/r/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (eddelbuettel) [reply apis/r/README.md]: Fixed in 1a1198c

**Review by johnkerl — CHANGES_REQUESTED** [review]
Here's what I found on a fresh EC2 Ubuntu instance.

My own criteria were: what it takes to install on a fresh system and be able to run unit-test code.

## System-level dependencies

```
sudo apt update
sudo apt install r-base-core
sudo apt install cmake
```

My ask: we should explicitly list these dependencies.

## Cloning the repo

```
$ git clone git@github.com:single-cell-data/TileDB-SOMA.git
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

But this worked:

```
git clone https://github.com/single-cell-data/TileDB-SOMA
```

My ask: we should mention both options for cloning.

## R dependencies

```
$ cd TileDB-SOMA/apis/r

> install.packages("remotes")
> remotes::install_deps(".")
```

This worked great.

## R build

```
$ R CMD build .
Error in loadVignetteBuilder(pkgdir, TRUE) :
  vignette builder 'knitr' not found
```

After I did this

```
> install.packages("knitr")
```

then the `R CMD build .` worked fine.

My ask: we should mention the `knitr` dependency, or suggest `R CMD build --no-build-vignettes --no-manual .`.

## R install

```
R CMD INSTALL tiledbsoma_*.tar.gz`
```

This worked great. Although I think (my ask here) along with the `cleanup` we should mention `rm -f tiledbsoma_*.tar.gz` before the install. (The `cleanup` script does not remove these.)

## Unit tests

```
> install.packages("testthat")
> testthat::test_dir('tests/testthat')
```

These failed with

```
Error (test_SOMADataFrame.R:6): Basic mechanics
Error in `eval(code, test_env)`: object 'SOMADataFrame' not found
```

If I did

```
> library(tiledbsoma)
> testthat::test_dir('tests/testthat')
```

then I got

```
Error (test_SOMAReader-Basics.R:10): Basic SOMAReader
Error in `arrow_to_dt(soma_reader(uri))`: could not find function "arrow_to_dt"
...
Error (test_SOMASparseNdArray.R:16): SOMASparseNDArray creation
Error in `is_arrow_table(tbl)`: could not find function "is_arrow_table"
...
```

etc.

I don't know what to ask for here. But it feels like (to me, as a relative R newcomer) that if I were going to do development, I'd want to build, install, and run unit tests. (Perhaps I'm doing something wrong here.)

**Comment by eddelbuettel** [comment]
I hear you.  None of what you write is wrong per se.  Alltogether it is too long and belongs into a vignette or other docs.  To me a README should be _crisp_ and _concise_.

**Comment by johnkerl** [comment]
Sorry to be clear @eddelbuettel I'm  not asking for my notes to be the entire content of the `README.md`

Crisp/concise things:

* We should list `cmake` as a dependency (one line)
* We should mention both URLs for `git clone` (one more line)
* We should list `knitr` as a dependency for the build (one more line), or rephrase the build command (a modification to one line)
* Unit tests -- I'm simply asking for help here

**Comment by eddelbuettel** [comment]
Your points 1 to 4 one by one

1.  Agreed, ditto `git`.  Our builds from source imply that.
2.  I suggest to only show `https://...`  It was my error to show 'my' git clone command as I have a ssh key req'ed
3. As an experienced R dev:  No.  That is implicit when you have vignettes.
4. As an experienced R dev:   We suggest what the R manuals and CRAN suggsted: `R CMD build ...` followed by `R CMD check` (which I omitted / switched to `R CMD INSTALL` instead)

**Review by johnkerl — APPROVED** [review]
🚢

**Comment by eddelbuettel** [comment]
Thanks for the comments!  They really helped.

**Pull request merged**